blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -44,13 +44,63 @@
 import java.util.TreeSet;
 import java.util.concurrent.TimeUnit;
 
-// keeps track of dirty slots in memory. Operations must be threadsafe.
-// todo: explore using ReadWrite locks (might not make a difference).
-
 /**
- * Each node is responsible for sharded slots (time ranges) of rollups.  This class keeps track of the execution of
- * those rollups and the states they are in.  
- * 
+ *
+ * The {@code ScheduleContext} class coordinates access to the states of slots.
+ * It mediates between the rollup service, the ingestion service, and the
+ * database.
+ * <p>
+ *
+ * The class's chief purpose is to coordinate access to slots' states and the
+ * list of slots that need to be re-rolled between {@link RollupService},
+ * {@link IngestionService}, and the database. There's several other classes
+ * and components involved, but those are the most important.
+ * <p>
+ *
+ * When ingestion services receive incoming metrics, they will change the
+ * associated slot's state to
+ * {@link com.rackspacecloud.blueflood.service.UpdateStamp.State#Active Active}
+ * (needs to be re-rolled) by calling {@link #update(long, int)}. That state
+ * information will be persisted to the database by the
+ * {@link ShardStatePusher} class. On the rollup nodes, the
+ * {@link ShardStatePuller} will read the state info into memory by calling
+ * {@link com.rackspacecloud.blueflood.service.ShardStateManager.SlotStateManager#updateSlotOnRead(SlotState)}.
+ * The rollup service will then identity slots that need to be re-rolled (by
+ * calling {@link #scheduleSlotsOlderThan(long)}), pick a slot to rollup and
+ * mark it as
+ * {@link com.rackspacecloud.blueflood.service.UpdateStamp.State#Running Running}
+ * (via {@link #getNextScheduled()} ), do the rollup, and then mark the slot as
+ * {@link com.rackspacecloud.blueflood.service.UpdateStamp.State#Rolled Rolled}
+ * (via {@link #clearFromRunning(SlotKey)}).
+ * <p>
+ *
+ * If doing the rollup fails for some reason, the rollup service will call
+ * {@link #pushBackToScheduled(SlotKey, boolean)}, to return the slot to the
+ * queue of slots to be rolled.
+ * <p>
+ *
+ * There are additional methods for managing shards ({@link #addShard(int)} and
+ * {@link #removeShard(int)}) which don't appear to be used.
+ * <p>
+ *
+ * There are also methods which provide information about the state of things
+ * (e.g. {@link #getScheduledCount()},
+ * {@link #getSlotStamps(Granularity, int)},
+ * {@link #getRecentlyScheduledShards()},
+ * {@link #getMetricsState(int, String, int)}) which either aren't used, are
+ * only used for testing, or are used to provide information to external system
+ * via JMX or yammer metrics.
+ * <p>
+ *
+ *
+ *
+ * keeps track of dirty slots in memory. Operations must be threadsafe.
```

Comment:
Can we capitalize the K?  I'm crazy, but while reading, since it was a sentence fragment, I thought the front half the sentence got cut off.  Confused me for a bit, especially since complete sentences are used above it.